logging/callback.go

```diff
@@ -0,0 +1,68 @@
+package logging
+
+// CallbackLogger is an implementation of Logger that forwards calls to Log and
+// Debug to a delegate callback. CallbackLogger MUST have LogTarget set.
+type CallbackLogger struct {
+	// LogTarget is the delegate used for log messages.
+	LogTarget Printf
+	// DebugTarget is the delegate used for debug messages. When this
+	// field isn't set, no debug logging will occur and IsDebug() will
+	// return false
+	DebugTarget Printf
+}
+
+// Printf is the function signature for fmt.Printf() and alike functions.
+type Printf func(fmt string, v ...interface{})
```

Comment:
```suggestion
type Callback func(fmt string, v ...interface{})
```

I prefer the name `Callback` for consistency with `CallbackLogger`.